pkg/apis/apps.kubermatic/v1/application_definition.go

```diff
@@ -224,6 +224,29 @@ type ApplicationDefinitionSpec struct {
 	// These settings can be overridden in applicationInstallation.
 	DefaultDeployOptions *DeployOptions `json:"defaultDeployOptions,omitempty"`
 
+	// +kubebuilder:validation:Pattern:=v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?
+	// +kubebuilder:validation:Type=string
+
+	// DefaultVersion of the application to use, if not specified the latest available version will be used.
+	// +optional
+	DefaultVersion string `json:"defaultVersion,omitempty"`
+
+	// Enforced specifies if the application is enforced to be installed on the user clusters. Enforced applications are
+	// installed/updated by KKP for the user clusters. Users are not allowed to update/delete them. KKP will revert the changes
+	// done by the application to the desired state specified in the ApplicationDefinition.
+	// +optional
+	Enforced bool `json:"enforced,omitempty"`
+
+	// Default specifies if the application should be installed by default when a new user cluster is created. Default applications are
+	// not enforced and users can update/delete them. KKP will only install them during cluster creation if the user didn't explicitly
+	// opt out from installing default applications.
+	// +optional
+	Default bool `json:"default,omitempty"`
+
+	// Selector is used to select the targeted user clusters for defaulting and enforcing applications. This is only used for default/enforced applications and ignored otherwise.
+	// +optional
+	Selector DefaultingSelector `json:"selector,omitempty"`
```

Comment:
For reviewer:

We can also call it `defaultingSelector` but I'd personally just prefer `selector` since it's more native; how similar resources work like deployments, etc.